Fix various paths to include MOZ_APP_VERSION

This has been committed upstream but was not included in
the 52.4.0 sources that spidermonkey was rolled from.

--- a/config/baseconfig.mk	2017-10-03 14:00:45.000000000 -0400
+++ b/config/baseconfig.mk	2017-10-03 16:36:10.857663794 -0400
@@ -2,10 +2,10 @@
 # directly in python/mozbuild/mozbuild/base.py for gmake validation.
 # We thus use INCLUDED_AUTOCONF_MK to enable/disable some parts depending
 # whether a normal build is happening or whether the check is running.
-includedir := $(includedir)/$(MOZ_APP_NAME)
-idldir = $(includedir)/idl/$(MOZ_APP_NAME)
-installdir = $(libdir)/$(MOZ_APP_NAME)
-sdkdir = $(libdir)/$(MOZ_APP_NAME)
+includedir := $(includedir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+idldir = $(includedir)/idl/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+installdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
+sdkdir = $(libdir)/$(MOZ_APP_NAME)-$(MOZ_APP_VERSION)
 ifeq (.,$(DEPTH))
 DIST = dist
 else
